src/Grid.hpp

```diff
@@ -0,0 +1,90 @@
+//
+//  Grid.hpp
+//  Game of Life
+//
+//  Created by Alex Frasca on 01/05/2023.
+//
+#pragma once
+#ifndef Grid_hpp
+#define Grid_hpp
+
+#include <SFML/Graphics.hpp>
+#include <vector>
+#include <cmath>
+#include <functional>
+
+namespace conway {
+
+//several values so we can draw nice colors as the cells die instead of just on/off
+enum CELL_STATE {
+    CELL_OFF = 0,
+    CELL_LOW = 1,
+    CELL_MEDIUM = 2,
+    CELL_HIGH = 3,
+    CELL_ON = 4
+};
+const float DEFAULT_SPACING = 16.0;
+
+struct Cell {
+    
+    Cell() {};
+    Cell(sf::Vector2<int> idx, sf::Vector2<float> pos, float spc, CELL_STATE st);
+    
+    void tick();
+    void setState();
+    // for a dead cell, use a frame-wise decrement for animation/visualization purposes
+    void decrement();
+    // only for when the size or spacing of the grid is changed, not for each tick
+    void updatePosition();
+    
+    // Cell member vars
+    sf::Vector2<int> index;
+    sf::Vector2<float> position;
+    // for drawing the cell
+    sf::ConvexShape drawable;
+    
+    float spacing;
```

Comment:
Variables should be initialized when they are defined (via `{}`).

Reply:
done.